cpp/src/arrow/array/builder_binary.h

```diff
@@ -487,18 +487,42 @@ class ARROW_EXPORT StringHeapBuilder {
 
   void SetBlockSize(int64_t blocksize) { blocksize_ = blocksize; }
 
-  void UnsafeAppend(BinaryViewType::c_type* v) {
-    memcpy(current_out_buffer_, v->raw.data, v->size());
-    v->io = {v->size(), v->raw.prefix, static_cast<int32_t>(blocks_.size() - 1),
-             current_offset_};
-    current_out_buffer_ += v->size();
-    current_remaining_bytes_ -= v->size();
-    current_offset_ += static_cast<int32_t>(v->size());
+  using c_type = BinaryViewType::c_type;
+
+  template <bool Safe>
+  std::conditional_t<Safe, Result<c_type>, c_type> Append(const uint8_t* value,
+                                                          int64_t length) {
```

Comment:
Might be a good idea to keep this private for now as this is not a pattern that extends to the other builders.

Reply:
Sorry, this is a helper class. Nvm.